### DISCUSSION TITLE
How exactly should one deploy a frontend inside a turborepo to Vercel?

### DISCUSSION THREAD

**Discussion by cording12** [vercel/turborepo#9148] [Help]
### Summary

I've been struggling to understand this and hoping someone can shed some light. I've spent a while reading Vercel and Turbo repo documentation but can't find an answer. 

Let's take the create-turbo [basic example.](https://github.com/vercel/turborepo/tree/main/examples/basic)

My current operating method to deploy the **web** app would be the following:

1. Create new project in Vercel dashboard and import the Git repository
2. Set the `Root Directory` to `apps/web`
3. Deploy

I've had success with this in the past. However, I also run into issues whereby Vercel can't find a custom package, such as a UI package. The builds fail, until I manually redeploy from the dashboard, ignore cache, and then it works.

I think I potentially narrowed this down to being a symlink issue with pnpm, mentioned in this issue [here](https://github.com/vercel/vercel/issues/8063). 

The part where I'm struggling to understand pertains to the Vercel documentation on [Root Directory](https://vercel.com/docs/deployments/configure-a-build#root-directory) which states: 

> Your app will not be able to access files outside of that directory. You also cannot use .. to move up a level

To me this seems to directly conflict with the deployment of a single app in a turbo monorepo which relies on packages outside the root directory as shown above. 

This now raises two issues for me:
1. Is this an issue with pnpm/yarn and symlinks in Vercel deployments?
2. Is this an issue with changing the root directory 
3. Other?

I can't find a good example of how exactly to deploy a turbo repo in such a manner - or am I fundamentally not understanding something here?

**Comment by anthonyshew** [comment]
The Root Directory settings you've described is correct. I've confirmed that the `basic` example deploys as expected. Do you have a reproduction of a repository you'd expect to be working that isn't?

**Comment by jhanstra** [comment]
I am very confused on this as well. @anthonyshew how should one go about deploying an app in a monorepo to Vercel manually from the command line? I haven't seen that demonstrated anywhere. Do I *need* to use Github integration with Vercel to do this? Every avenue I've tried seems to be a dead-end.
